simulators/portal-bridge-tests/src/main.rs

```diff
@@ -0,0 +1,180 @@
+
+    let mut suite = Suite {
+        name: "portal-bridge-tests".to_string(),
+        description: "The portal bridge test suite".to_string(),
+        tests: vec![],
+    };
+
+    suite.add(TestSpec {
+        name: "Portal bridge tests".to_string(),
+        description: "".to_string(),
+        always_run: false,
+        run: test_portal_bridge,
+        client: None,
+    });
+
+    let sim = Simulation::new();
+    run_suite(sim, suite).await;
+}
+
+async fn run_suite(host: Simulation, suite: Suite) {
+    let name = suite.clone().name;
+    let description = suite.clone().description;
+
+    let suite_id = host.start_suite(name, description, "".to_string()).await;
+
+    for test in &suite.tests {
+        test.run_test(host.clone(), suite_id, suite.clone()).await;
+    }
+
+    host.end_suite(suite_id).await;
+}
+
+dyn_async! {
+   async fn test_portal_bridge<'a> (test: &'a mut Test, _client: Option<Client>) {
+        // Get all available portal clients
+        let clients = test.sim.client_types().await;
+
+        let private_key_1 = "fc34e57cc83ed45aae140152fd84e2c21d1f4d46e19452e13acc7ee90daa5bac".to_string();
+        let private_key_2 = "e5add57dc4c9ef382509e61ce106ec86f60eb73bbfe326b00f54bf8e1819ba11".to_string();
+        let private_key_3 = "80449579bd9d09a03d245b0a018f36887a0e97d3d2994d89841e9f5bd5bfedf0".to_string();
+
+        // Iterate over all possible pairings of clients and run the tests (including self-pairings)
+        for ((client_a, client_b), client_c) in clients.iter().cartesian_product(clients.iter()).cartesian_product(clients.iter()) {
+            test.run(
+                NClientTestSpec {
+                    name: format!("Bridge test. A:{} --> B:{} --> C:{}", client_a.name, client_b.name, client_c.name),
+                    description: "".to_string(),
+                    always_run: false,
+                    run: test_bridge,
+                    private_keys: Some(vec![Some(private_key_1.clone()), Some(private_key_2.clone()), Some(private_key_3.clone())]),
+                    clients: vec![client_a.clone(), client_b.clone(), client_c.clone()],
+                }
+            ).await;
+        }
+   }
+}
+
+dyn_async! {
+    async fn test_bridge<'a> (clients: Vec<Client>) {
+        let (client_a, client_b, client_c) = match clients.iter().collect_tuple() {
+            Some((client_a, client_b, client_c)) => (client_a, client_b, client_c),
+            None => {
+                panic!("Unable to get expected amount of clients from NClientTestSpec");
+            }
+        };
+
+        let master_acc = MasterAccumulator::default();
+        let header_oracle = HeaderOracle::new(master_acc);
+        let portal_clients = vec![client_a.rpc.clone()];
+        let epoch_acc_path = "validation_assets/epoch_acc.bin".into();
+        let mode = BridgeMode::Test("./test-data/test_data_collection_of_forks_blocks.json".into());
+        let pandaops_middleware = PandaOpsMiddleware::default();
```

Comment:
This is something awfully specific in this test. Does it mean you need to have access to this infrastructure to be able to run this test locally?

Imo 1 crucial requirement for portal-hive tests is that anybody should be able to rerun the tests locally.

Reply:
You don't need access to any infastructure. PandOs's uses fake values, this is just some weirdness. This can be ran locally to run test data from files.

So yes anybody can run this locally :D

Reply:
Perhaps add a little comment there to clarify this weirdness?